app/src/main/java/com/example/filemanager/AllFilesFragment.kt

```diff
@@ -59,4 +71,65 @@ class AllFilesFragment : Fragment(), ItemClickListener {
     override fun onItemClickListener() {
         Toast.makeText(context, "Item clicked", Toast.LENGTH_SHORT).show()
     }
+
+    private fun showPermissionDialog() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+            try {
+                val intent = Intent(Settings.ACTION_MANAGE_ALL_FILES_ACCESS_PERMISSION)
+                intent.addCategory("android.intent.category.DEFAULT")
+                intent.data = Uri.parse(String.format("package:%s", arrayOf<Any>(applicationContext().packageName)))
+                activity?.startActivityForResult(intent, 2000)
+            } catch (e: Exception) {
+                val intent = Intent(Settings.ACTION_MANAGE_ALL_FILES_ACCESS_PERMISSION)
+                activity?.startActivityForResult(intent, 2000)
+            }
+        } else ActivityCompat.requestPermissions(
+            super.requireActivity(),
+            arrayOf(Manifest.permission.WRITE_EXTERNAL_STORAGE, Manifest.permission.READ_EXTERNAL_STORAGE),
+            333
+        )
+    }
+
+    private fun checkPermission(): Boolean {
+        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+            Environment.isExternalStorageManager()
+        } else {
+            val write = ContextCompat.checkSelfPermission(
+                applicationContext(),
+                Manifest.permission.WRITE_EXTERNAL_STORAGE
+            )
+            val read = ContextCompat.checkSelfPermission(
+                applicationContext(),
+                Manifest.permission.READ_EXTERNAL_STORAGE
+            )
+            write == PackageManager.PERMISSION_GRANTED &&
+                    read == PackageManager.PERMISSION_GRANTED
+        }
+    }
+
+    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<String?>, grantResults: IntArray) {
```

Comment:
Does this method gets called back when you grant the permissions? it's not being called for me, I had to override it in the `MainActivity`. This is strange 🤔 .

Reply:
I also had to override it in MainActivity, but it is not called